File: week2/unique-morse-code-words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        HashSet<String> set = new HashSet<>();
        for (String word : words) {
            set.add(transform(word));
        }
        return set.size();
    }
    
    public String transform(String word) {
        String[] morse = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < word.length(); i++) {
            sb.append(morse[(int) word.charAt(i) - 97]);
        }
        return sb.toString();
    }
}

File: 2026/week2/unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        transformations = {}
        for word in words:
            morse_word = self.morsify(word)
            if morse_word in transformations:
                transformations[morse_word] += 1
            else:
                transformations[morse_word] = 1
        return len(transformations)

    def morsify(self, word: str) -> str:
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        res = []
        for c in word:
            res.append(morse[ord(c) - 97])
        return "".join(res)
